### PR TITLE
Update documentation to keras reuters dataset

### PR DESCRIPTION
**Pull request by hxy9243**
From discussions and references from:

- https://github.com/keras-team/keras/issues/12072#issuecomment-458154097
- https://martin-thoma.com/nlp-reuters/

Add documentation:

- That explains the word indices returned from keras `keras.datasets.reuters.get_word_index`.
- Add helper function to return `ylabels` for label data.

### CODE REVIEW

File: keras/datasets/reuters.py [hunk 1]
```diff
@@ -164,6 +164,12 @@ def load_data(
 @keras_export("keras.datasets.reuters.get_word_index")
 def get_word_index(path="reuters_word_index.json"):
     """Retrieves a dict mapping words to their index in the Reuters dataset.
+    Actual word indices starts from 3, with 3 indices reserved for:
```
Comment (fchollet) [inline keras/datasets/reuters.py:167]: Add line break above

File: keras/datasets/reuters.py [hunk 2]
```diff
@@ -182,3 +188,62 @@ def get_word_index(path="reuters_word_index.json"):
     )
     with open(path) as f:
         return json.load(f)
+
+
+@keras_export("keras.datasets.reuters.get_ylabels")
+def get_ylabels():
```
Comment (fchollet) [inline keras/datasets/reuters.py:194]: `get_label_names`

File: keras/datasets/reuters.py [hunk 3]
```diff
@@ -182,3 +188,62 @@ def get_word_index(path="reuters_word_index.json"):
...
+    """Returns the y label as a list of strings with indices matching
+    training data.
```
Comment (fchollet) [inline keras/datasets/reuters.py:196]: This should fit in 1 line

File: keras/datasets/reuters.py [hunk 4]
```diff
@@ -182,3 +188,62 @@ def get_word_index(path="reuters_word_index.json"):
...
+
+    See references from:
+    - https://github.com/keras-team/keras/issues/12072#issuecomment-458154097
```
Comment (fchollet) [inline keras/datasets/reuters.py:199]: Use markdown for links and add line break before list

File: keras/datasets/reuters.py [hunk 5]
```diff
@@ -164,6 +164,12 @@ def load_data(
 @keras_export("keras.datasets.reuters.get_word_index")
 def get_word_index(path="reuters_word_index.json"):
     """Retrieves a dict mapping words to their index in the Reuters dataset.
+    Actual word indices starts from 3, with 3 indices reserved for:
+    0 (padding), 1 (start), 2 (oof).
+
+    E.g. word index of 'the' is 1, but the in the actual training data, the
+    index of 'the' will be 1+3 = 4. Vice versa, to translate word indices in
```
Comment (fchollet) [inline keras/datasets/reuters.py:171]: Add spaces around +

File: keras/datasets/reuters.py [hunk 6]
```diff
@@ -164,6 +164,12 @@ def load_data(
 @keras_export("keras.datasets.reuters.get_word_index")
 def get_word_index(path="reuters_word_index.json"):
     """Retrieves a dict mapping words to their index in the Reuters dataset.
+    Actual word indices starts from 3, with 3 indices reserved for:
+    0 (padding), 1 (start), 2 (oof).
```
Comment (fchollet) [inline keras/datasets/reuters.py:168]: OOV

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/datasets/reuters.py [hunk 7]
```diff
@@ -182,3 +189,62 @@ def get_word_index(path="reuters_word_index.json"):
     )
     with open(path) as f:
         return json.load(f)
+
+
+@keras_export("keras.datasets.reuters.get_ylabels")
```
Comment (fchollet) [inline keras/datasets/reuters.py:194]: Update this line

File: keras/datasets/reuters.py [hunk 8]
```diff
@@ -182,3 +189,62 @@ def get_word_index(path="reuters_word_index.json"):
...
+def get_label_names():
+    """Returns labels as a list of strings with indices matching training data.
+
+    See references from:
+
+    - [Github Discussion](https://github.com/keras-team/keras/issues/12072#issuecomment-458154097)
```
Comment (fchollet) [inline keras/datasets/reuters.py:200]: You can remove this link

File: keras/datasets/reuters.py [hunk 9]
```diff
@@ -182,3 +189,62 @@ def get_word_index(path="reuters_word_index.json"):
     )
     with open(path) as f:
         return json.load(f)
+
+
+@keras_export("keras.datasets.reuters.get_ylabels")
+def get_label_names():
+    """Returns labels as a list of strings with indices matching training data.
+
+    See references from:
```
Comment (fchollet) [inline keras/datasets/reuters.py:198]: `Reference:`

**Review by fchollet — APPROVED** [review]
LGTM, thank you

**Comment by hxy9243** [comment]
Thanks very much for the feedback and review!
